### PR TITLE
Code files for Type Hints tutorial

### PR DESCRIPTION
**Pull request by claudian37**
Adding sample code files for tutorial on [How to Use Type Hints for Multiple Return Types in Python](https://realpython.com/python-type-hints-multiple-types/)

### CODE REVIEW

File: python-type-hints-multiple-types/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+
```
Comment (bzaczynski) [inline python-type-hints-multiple-types/README.md:1]: There's an extra blank line that we can remove from here.

File: python-type-hints-multiple-types/01_type_hints_for_one_piece_of_data.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+
+def parse_email(email_address: str) -> str | None:
+    if "@" in email_address:
+        username, domain = email_address.split("@")
+        return username
+    return None
+
+
+def parse_email(email_address: str) -> Union[str, None]:
```
Comment (bzaczynski) [inline python-type-hints-multiple-types/01_type_hints_for_one_piece_of_data.py:11]: The linter complains about redefining a function. Maybe you can give the function a different name or put it in a comment like so:

```python
# def parse_email(email_address: str) -> Union[str, None]:
#     if "@" in email_address:
#         username, domain = email_address.split("@")
#         return username
#     return None
```

File: python-type-hints-multiple-types/02_type_hints_for_multiple_pieces_of_data.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+
+def parse_email(email_address: str) -> tuple[str, str] | None:
+    if "@" in email_address:
+        username, domain = email_address.split("@")
+        return username, domain
+    return None
+
+
+def parse_email(email_address: str) -> Union[Tuple[str, str], None]:
```
Comment (bzaczynski) [inline python-type-hints-multiple-types/02_type_hints_for_multiple_pieces_of_data.py:11]: Another redefined function.

File: python-type-hints-multiple-types/05_type_hints_for_generator.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+
+def parse_emails(emails: list[str]) -> Iterator[tuple[str, str]]:
+    for email in emails:
+        if "@" in email:
+            username, domain = email.split("@")
+            yield username, domain
+
+
+def parse_emails(emails: Iterable[str]) -> Iterable[tuple[str, str]]:
```
Comment (bzaczynski) [inline python-type-hints-multiple-types/05_type_hints_for_generator.py:32]: Flagging yet another redefined function.

File: python-type-hints-multiple-types/06_type_aliases.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+from typing import TypeAlias
+
+
+EmailComponents = tuple[str, str] | None
```
Comment (bzaczynski) [inline python-type-hints-multiple-types/06_type_aliases.py:4]: Do we need to include the old-style way of defining an alias? Maybe only the new one will suffice?

```python
EmailComponents: TypeAlias = tuple[str, str] | None
```

File: python-type-hints-multiple-types/README.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+### 04_type_hints_for_factory_function
+
+
+### 05_type_hints_for_generator
+
+
+### 06_type_aliases
+
+
+### 07_static_type_checking
```
Comment (bzaczynski) [inline python-type-hints-multiple-types/README.md:40]: If you're unsure what to put in the README file, then I'd say let's cut the highlighted lines.

File: python-type-hints-multiple-types/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,4 @@
+mypy==1.5.1
+mypy-extensions==1.0.0
+tomli==2.0.1
```
Comment (bzaczynski) [inline python-type-hints-multiple-types/requirements.txt:3]: I believe, we don't need this dependency, which is a backport of the standard library module `tomllib` introduced in  Python 3.11.

**Review by bzaczynski — CHANGES_REQUESTED** [review]
@claudian37 Hey Claudia! Thanks for creating a separate PR in the materials repository. Please find a few comments and let me know if you have further questions.

**Comment by claudian37** [comment]
> @claudian37 Hey Claudia! Thanks for creating a separate PR in the materials repository. Please find a few comments and let me know if you have further questions.

I've made the suggested changes. I will be traveling until Friday, so probably won't be able to make any changes before then. Thanks for reviewing my PRs!

**Review by bzaczynski — APPROVED** [review]
@claudian37 Thanks for the updates!

**Pull request merged**